### PR TITLE
Development

### PR DESCRIPTION
**Pull request by cneker**

### CODE REVIEW

File: InnoGotchi.MVC/InnoGotchi.MVC/Services/AuthenticationService.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            _httpClientFactory = httpClientFactory;
+            _mapper = mapper;
+        }
+
+        public async Task<UserForAuthenticationDto> RegisterUserAsync(UserForRegistrationDto userForReg, ModelStateDictionary modelState)
+        {
+            var httpRequestMessage = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Post,
+                RequestUri = new Uri("https://localhost:7208/api/users"),
```
Comment (kseniya-kalita) [inline InnoGotchi.MVC/InnoGotchi.MVC/Services/AuthenticationService.cs]: Url should be stored in appsettings

File: InnoGotchi.MVC/InnoGotchi.MVC/Services/FarmService.cs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            var jsonContent = await httpResponseMessage.Content.ReadAsStringAsync();
+
+            var freindsFarms = JsonConvert.DeserializeObject<IEnumerable<FarmOverviewDto>>(jsonContent);
+
+            return freindsFarms;
+        }
+
+        public async Task<FarmOverviewDto> GetUserFarmOverviewAsync(string jwt, string userId)
+        {
+            var httpRequestMessage = new HttpRequestMessage()
```
Comment (kseniya-kalita) [inline InnoGotchi.MVC/InnoGotchi.MVC/Services/FarmService.cs]: The code with creating, sending and processing response using http client is repeated. You should write some wrapper for http client  to reuse the logic. Example can be found on the screenshot and Internet
image

File: InnoGotchi.MVC/InnoGotchi.MVC/Middlewares/AddingAuthenticationHeaderMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+        public AddingAuthenticationHeaderMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var token = context.Request.Cookies["jwt"];
+            if (token != null)
```
Comment (kseniya-kalita) [inline InnoGotchi.MVC/InnoGotchi.MVC/Middlewares/AddingAuthenticationHeaderMiddleware.cs:15]: Trying to register, received this error
image

File: InnoGotchi.MVC/InnoGotchi.MVC/Controllers/FarmController.cs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        public FarmController(IFarmService farmService, IPetService petService, IMapper mapper)
+        {
+            _farmService = farmService;
+            _petService = petService;
+            _mapper = mapper;
+        }
+
+        public async Task<IActionResult> FarmOverview()
+        {
+            var jwt = Request.Cookies["jwt"];
```
Comment (kseniya-kalita) [inline InnoGotchi.MVC/InnoGotchi.MVC/Controllers/FarmController.cs:24]: extract this two lines in a separate method as they're repeated

**Review by kseniya-kalita — COMMENTED** [review]
Please fix the error with registering for the first time as it's not possible to use the app for now. Review and fix the comments. Please also take a look at this doc with code style requirements
https://docs.google.com/document/d/1Z-Y6klf1vIUKT0CUoZseBUTz18MndJejaAOEB5u1w3c/edit#heading=h.kr1lw5fd72l6

File: InnoGotchi.MVC/InnoGotchi.MVC/Services/AuthenticationService.cs (hunk 1 above)
Comment (cneker) [reply InnoGotchi.MVC/InnoGotchi.MVC/Services/AuthenticationService.cs]: updated

File: InnoGotchi.MVC/InnoGotchi.MVC/Services/FarmService.cs (hunk 2 above)
Comment (cneker) [reply InnoGotchi.MVC/InnoGotchi.MVC/Services/FarmService.cs]: made

File: InnoGotchi.MVC/InnoGotchi.MVC/Middlewares/AddingAuthenticationHeaderMiddleware.cs (hunk 3 above)
Comment (cneker) [reply InnoGotchi.MVC/InnoGotchi.MVC/Middlewares/AddingAuthenticationHeaderMiddleware.cs:15]: i don't really know how to recreate this error. i tried to register new user on my machine and there was no exception. i also cloned these two projects to a new pc, made some configuration (just set up Environment variable) and there was still no error. i don't really know what to do

File: InnoGotchi.MVC/InnoGotchi.MVC/Controllers/FarmController.cs (hunk 4 above)
Comment (cneker) [reply InnoGotchi.MVC/InnoGotchi.MVC/Controllers/FarmController.cs:24]: made

**Pull request closed without merging**